Working HEROLight

CoryNessCTR opened the pull request:
HEROLight works & Project names got changed from CandleLight/Project to HEROLight
This will fix #1

CoryNessCTR committed:
HEROLight works

HEROLight is tested and works with various tests.
Key was to reduce the number of CAN Queue Size from 16 to 4 to allocate enough memory for the Queues.

Fixes https://github.com/CrossTheRoadElec/HERO-STM32F4/issues/1

CoryNessCTR committed:
Renamed Project to HEROLight & Deleted Project.uvoptx

CoryNessCTR committed:
CANdleLight Directory renamed to HEROLight

Renamed directory to HEROLight as well as update gitignore.

ozrien commented on Projects/STM32F429VI-HERO/Applications/USB_Device/HEROLight/Inc/dfu.h:
```diff
@@ -24,6 +24,9 @@ THE SOFTWARE.
 */
 
-#pragma once
+#ifndef DFU_H
```
see why this was necessary

ozrien commented on Projects/STM32F429VI-HERO/Applications/USB_Device/HEROLight/Inc/gpio.h:
```diff
@@ -24,6 +24,9 @@ THE SOFTWARE.
 */
 
-#pragma once
+#ifndef GPIO_H
```
see why this was necessary

ozrien commented on Projects/STM32F429VI-HERO/Applications/USB_Device/HEROLight/Inc/timer.h:
```diff
@@ -24,9 +24,13 @@ THE SOFTWARE.
 */
 
-#pragma once
+#ifndef TIMER_H
```
see why this was necessary

ozrien committed:
Delete startup_stm32f429xx.lst

This is an output file, not needed.

ozrien commented on Projects/STM32F429VI-HERO/Applications/USB_Device/HEROLight/MDK-ARM/HEROLight.uvprojx:
```diff
@@ -530,14 +530,14 @@
             </File>
             <File>
-              <FileName>stm32f4xx_hal_i2c.c</FileName>
+              <FileName>stm32f4xx_hal_pcd.c</FileName>
```
put these back, avoid superfluous changes to project files.

CoryNessCTR committed:
USB A And Mini Additions

Renamed first project to HERO_FS_USB_A to make it clear that it uses the A port.
Added second project called HERO_HS_USB_MINI that works with the Mini Port.
Modified usbd_conf.c to switch between fs and hs based on preprocessor define.
Modified stm32f4xx and usbd_conf.c hpcd_USB_FS to hpcd_USB.
added 2 interrupt functions for HS interrupts.

CoryNessCTR committed:
Header Guards replaced with pragma once's

Header guards aren't needed by the compiler to compile, so they were replaced with pragma once's to be more similar to CandleLight_fw
